### PR TITLE
Text & Paragraph Usage Criteria + Text CSS Alignment

### PR DESCRIPTION
**Pull request by marcod1419**
This updates both the `Text` and `Paragraph` docs to make their usage criteria more clear. Additionally, I made all typography components point to the shared typography scss file. (Before, every component _except_ for the `Text` component was pointing to a stylesheet in `Text`'s folder.

### CODE REVIEW

File: packages/ChevronLink/ChevronLink.modules.scss [hunk 1]
```diff
@@ -15,7 +15,7 @@
 }
 
 .link {
-  composes: medium from '../Text/Text.modules.scss';
+  composes: medium from '../../shared/styles/Typography/Typography.modules.scss';
```
Comment (jraff) [inline packages/ChevronLink/ChevronLink.modules.scss]: Can we add `@tds/shared-typography` to ChevronLink dev dependencies?

File: packages/PriceLockup/PriceLockup.modules.scss [hunk 2]
```diff
@@ -66,11 +66,11 @@
 }
 
 .mediumText {
-  composes: medium from '../Text/Text.modules.scss';
+  composes: medium from '../../shared/styles/Typography/Typography.modules.scss';
```
Comment (jraff) [inline packages/PriceLockup/PriceLockup.modules.scss]: Can we add `@tds/shared-typography` to PriceLockup dev dependencies?

File: shared/components/BaseButton/BaseButton.modules.scss [hunk 3]
```diff
@@ -13,7 +13,7 @@ $button-text-color: $color-white;
 
 .base {
   composes: none rounded from '../../styles/Borders.modules.scss';
-  composes: medium boldFont from '../../../packages/Text/Text.modules.scss';
```
Comment (jraff) [inline shared/components/BaseButton/BaseButton.modules.scss]: Let's add`@tds/shared-typography` as a devDependency of BaseButton

File: shared/components/FormField/FormField.modules.scss [hunk 4]
```diff
@@ -21,7 +21,7 @@
   composes: thin rounded from '../../styles/Borders.modules.scss';
 
   composes: font from '../../styles/Forms.modules.scss';
-  composes: medium mediumFont color from '../../../packages/Text/Text.modules.scss';
+  composes: medium mediumFont color from '../../../shared/styles/Typography/Typography.modules.scss';
```
Comment (jraff) [inline shared/components/FormField/FormField.modules.scss]: Let's add`@tds/shared-typography` as a devDependency of FormField

**Comment by marcod1419** [comment]
@jraff Should be good to go now.

File: packages/Text/Text.jsx [hunk 5]
```diff
@@ -19,11 +19,12 @@ const Text = ({ block, bold, size, invert, children, ...rest }, context) => {
...
     styles[size],
     bold ? styles.boldFont : styles[`${size}Font`],
-    context.inheritColor ? styles.inheritColor : textColor
+    context.inheritColor ? styles.inheritColor : textColor,
+    block ? styles.blockText : null
   )
 
   return React.createElement(
-    block ? 'div' : 'span',
+    'span',
```
Comment (theetrain) [inline packages/Text/Text.jsx]: Since it'll always be one kind of tag, using JSX is preferred.

File: packages/Text/Text.md [hunk 6]
```diff
@@ -1,14 +1,25 @@
+### Minimal usage
+
 ```jsx
 <Text>Tell us what you think</Text>
 ```
 
-By default, the Text component will inherit font properties from its parent. Use the props to override this behaviour.
+### Usage criteria
+
+- Only used to modify text in `Paragraph`.
```
Comment (theetrain) [inline packages/Text/Text.md]: Let's rephrase this to say 'Should only be used to modify text in `Paragraph` as there are exceptions such as PriceLockup, tertiaryText in ExpandCollapse, and other standoffish text.
